### PR TITLE
Adding support for theremini SAO remote keyboard

### PR DESCRIPTION
**Pull request by hamster**
Supports using the theremini as a remote keyboard.
Mode selection is abstracted to allow multiple sources to change the mode

### CODE REVIEW

File: Software/Makefile [hunk 1]
```diff
@@ -131,6 +131,7 @@ SRC_FILES += \
   $(PROJ_DIR)/modules/led.cpp \
   $(PROJ_DIR)/modules/ws2812b.cpp \
   $(PROJ_DIR)/modules/VL6180X.cpp \
+  $(PROJ_DIR)/modules/SAO.cpp \
```
Comment (rlucero) [inline Software/Makefile:138]: You can exclude this file so it doesn't cause conflicts with people who have customized their Makefile (like me on Windoze).

**Comment by hamster** [comment]
Adding a bootloader and a uf2 conversion script

**Comment by hamster** [comment]
Adds a target for flashing bootloader from the Makefile - 'make flash_bootloader'

**Review by rlucero — CHANGES_REQUESTED** [review]

**Review by rlucero — APPROVED** [review]
Make files - you're on your own.

**Pull request merged**